instrumentation/spring/spring-kafka-2.7/testing/src/main/java/io/opentelemetry/testing/AbstractSpringKafkaNoReceiveTelemetryTest.java

```diff
@@ -0,0 +1,220 @@
+                                equalTo(SemanticAttributes.MESSAGING_OPERATION, "process"),
+                                satisfies(
+                                    SemanticAttributes.MESSAGING_MESSAGE_PAYLOAD_SIZE_BYTES,
+                                    AbstractLongAssert::isNotNegative),
+                                satisfies(
+                                    SemanticAttributes.MESSAGING_KAFKA_PARTITION,
+                                    AbstractLongAssert::isNotNegative)),
+                    span -> span.hasName("consumer").hasParent(trace.getSpan(2))));
+  }
+
+  @Test
+  void shouldCreateSpansForBatchReceiveAndProcess() throws InterruptedException {
+    Map<String, String> batchMessages = new HashMap<>();
+    batchMessages.put("10", "testSpan1");
+    batchMessages.put("20", "testSpan2");
+    sendBatchMessages(batchMessages);
+
+    AtomicReference<SpanData> producer1 = new AtomicReference<>();
+    AtomicReference<SpanData> producer2 = new AtomicReference<>();
+
+    testing()
+        .waitAndAssertSortedTraces(
+            orderByRootSpanKind(SpanKind.INTERNAL, SpanKind.CONSUMER),
+            trace -> {
+              trace.hasSpansSatisfyingExactly(
+                  span -> span.hasName("producer"),
+                  span ->
+                      span.hasName("testBatchTopic send")
+                          .hasKind(SpanKind.PRODUCER)
+                          .hasParent(trace.getSpan(0))
+                          .hasAttributesSatisfyingExactly(
+                              equalTo(SemanticAttributes.MESSAGING_SYSTEM, "kafka"),
+                              equalTo(SemanticAttributes.MESSAGING_DESTINATION, "testBatchTopic"),
+                              equalTo(SemanticAttributes.MESSAGING_DESTINATION_KIND, "topic")),
+                  span ->
+                      span.hasName("testBatchTopic send")
+                          .hasKind(SpanKind.PRODUCER)
+                          .hasParent(trace.getSpan(0))
+                          .hasAttributesSatisfyingExactly(
+                              equalTo(SemanticAttributes.MESSAGING_SYSTEM, "kafka"),
+                              equalTo(SemanticAttributes.MESSAGING_DESTINATION, "testBatchTopic"),
+                              equalTo(SemanticAttributes.MESSAGING_DESTINATION_KIND, "topic")));
+
+              producer1.set(trace.getSpan(1));
+              producer2.set(trace.getSpan(2));
+            },
+            trace ->
+                trace.hasSpansSatisfyingExactly(
+                    span ->
+                        span.hasName("testBatchTopic process")
+                            .hasKind(SpanKind.CONSUMER)
+                            .hasNoParent()
+                            .hasLinksSatisfying(
+                                links(
+                                    producer1.get().getSpanContext(),
+                                    producer2.get().getSpanContext()))
```

Comment:
OTLP drops the `isRemote` flag on span context, which makes the agent vs library instrumentation tests behave differently: agent tests have a `agent CL -> OTLP data -> text CL` conversion that clears out the flag, while in library instrumentation tests the value of the flag is preserved, and simply using `producer1.get().getSpanContext()` here makes the test fail (cause it's not a remote span).

@anuraaga @jack-berg should we perhaps modify the assertions so that they ignore the flag? Or at least provide an easy possibility to do so? WDYT?

Reply:
https://github.com/open-telemetry/oteps/pull/182